check_no_globals_test.go

```diff
@@ -1,179 +1,31 @@
-package main
+package gochecknoglobals
 
 import (
-	"fmt"
+	"flag"
 	"testing"
+
+	"golang.org/x/tools/go/analysis/analysistest"
 )
 
 func TestCheckNoGlobals(t *testing.T) {
-	cases := []struct {
```

Comment:
Since there's no longer any code to traverse directories, open files and do all the work before getting to the ast I didn't keep them. It felt bad to keep the code or write new code for this kind of job when it's handled by the analysis package already and won't be a part of the code.

To address what you mention in the comment - we could keep the testdata structure and move each folder into `testdata/src` then add 10 different package tests. However, some of them will be irrelevant since their only purpose was to test the file and directory walking.

I'll take a pass and start looking into how I can minimise the diff.